### PR TITLE
Adding functional test for Telegraf reporter.

### PR DESCRIPTION
**Pull request by nivancevic**

### CODE REVIEW

File: cassandra-diagnostics-ft/src/functional-test/resources/telegraf/cassandra-diagnostics.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+reporters:
+  - reporter: io.smartcat.cassandra.diagnostics.reporter.RiemannReporter
+    options:
+      riemannHost: riemann
+      riemannPort: 5555
+      batchEventSize: 1
+      
+      reporter: io.smartcat.cassandra.diagnostics.reporter.RiemannReporter
```
Comment (nbozic-zz) [inline cassandra-diagnostics-ft/src/functional-test/resources/telegraf/cassandra-diagnostics.yml]: Why do you use Riemann reporter in telegraf functional test?

File: cassandra-diagnostics-ft/src/functional-test/resources/telegraf/cassandra-env.sh [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+#MX4J_ADDRESS="-Dmx4jaddress=127.0.0.1"
+#MX4J_PORT="-Dmx4jport=8081"
+
+JVM_OPTS="$JVM_OPTS $MX4J_ADDRESS"
+JVM_OPTS="$JVM_OPTS $MX4J_PORT"
+JVM_OPTS="$JVM_OPTS $JVM_EXTRA_OPTS"
+
+JVM_OPTS="$JVM_OPTS -javaagent:/cassandra-diagnostics/cassandra-diagnostics-core.jar"
+JVM_OPTS="$JVM_OPTS -Dcassandra.diagnostics.config=/etc/cassandra/cassandra-diagnostics.yml"
+CLASSPATH="/cassandra-diagnostics/cassandra-diagnostics-connector.jar:/cassandra-diagnostics/cassandra-diagnostics-reporter-riemann.jar:$CLASSPATH"
```
Comment (nbozic-zz) [inline cassandra-diagnostics-ft/src/functional-test/resources/telegraf/cassandra-env.sh]: Again why riemann reporter, shouldn't this be telegraf reporter?

**Review by nbozic-zz — COMMENTED** [review]
Puzzled how does this work, or maybe I am missing something. You have in configuration Telegraf image, InfluxDB image and Cassandra image with diagnostics and in configuration you do not have telgraf reporter at all, only riemann reporter. 

To me it looks like either wrong configuration is somehow created or something like that.

File: cassandra-diagnostics-ft/src/functional-test/resources/telegraf/cassandra-diagnostics.yml (hunk 1 above)
Comment (mgobec) [inline cassandra-diagnostics-ft/src/functional-test/resources/telegraf/cassandra-diagnostics.yml]: Where is this reporter in configuration structure?

File: cassandra-diagnostics-ft/src/functional-test/resources/telegraf/telegraf.conf [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  ## Set the user agent for HTTP POSTs (can be useful for log differentiation)
+  # user_agent = "telegraf"
+  ## Set UDP payload size, defaults to InfluxDB UDP Client default (512 bytes)
+  # udp_payload = 512
+
+###############################################################################
+#                            INPUT PLUGINS                                    #
+###############################################################################
+
+[[inputs.cpu]]
```
Comment (mgobec) [inline cassandra-diagnostics-ft/src/functional-test/resources/telegraf/telegraf.conf]: Why read cpu and other os stats when we are testing telegraf reporter?

File: cassandra-diagnostics-ft/src/functional-test/resources/telegraf/cassandra-diagnostics.yml (hunk 1 above)
Comment (mgobec) [reply cassandra-diagnostics-ft/src/functional-test/resources/telegraf/cassandra-diagnostics.yml]: Same issues here (copy/paste I guess)
https://github.com/smartcat-labs/cassandra-diagnostics/blob/146074dee8c581fc22a2ddcb9785c520b18e93a7/cassandra-diagnostics-ft/src/functional-test/resources/riemann/cassandra-diagnostics.yml#L8

**Pull request merged**